### DISCUSSION TITLE
Organize homepage in some files

### DISCUSSION THREAD

**Discussion by 50l3r** [nuxt/nuxt#16234] [Questions]
Hi guys,

I use Nuxt3 and i have a homepage (index.vue) with >750 lines of code and i want to organize into multiple files.

What's the best way to organize big apps? I think that's not the best idea to create component if i only use in the  homepage.

- Do I modify the automatic router?
- Do I create an additional folder in /src and handle it from there?

I think the best method is to have a folder structure like the following:

```
src/
├─ pages/
│  ├─ index/
│  │  ├─ _partials/
│  │  │  ├─ section1.vue
│  │  │  ├─ section2.vue
│  │  │  ├─ section3.vue
│  │  ├─ index.vue
│  ├─ about/
│  │  ├─ index.vue
```

Any ideas of how can i do this?

Thanks in advance :)

**Comment by TheAlexLichter** [comment]
Why wouldn't you use components instead, which is the normal approach? 🤔

You can also take a look at a more "modular" approach and use `extends`, see also https://github.com/nuxt/framework/pull/9405

**Reply by 50l3r** [reply]
Personally i prefer divide usual components and specific component pages for better organization. Maybe extends its the way
